### PR TITLE
added TOS module

### PR DESCRIPTION
**Pull request by tobiasd**
If enabled the Terms of Service will be available at /tos, it is then linked from the registration form and the /friendica page. TOS text is free-form and can be spiced with BBCode.

Optionally a *privacy statement* can be shown on the page that should give some transparency.

Additionally there is a small fix for the BBCode parsing from the register_text.

And finally the messages.po was regenerated to include the new strings for the translation pipe.

### CODE REVIEW

**Comment by AndyHee** [comment]
I tested this PR and it works from what I can tell.

I suggest a minor change to the wording for clarity:

'...the user has to provide a display name (pen name), a nickname...' -->  '...the user has to provide a display name (pen name), a username (nickname)...'

plus a missing apostrophe:

'...nodes user...' --> '...node's user...'

and a possible comma:

'...by any visitor of the page even if other profile details are not displayed...' -> '...by any visitor of the page, even if other profile details are not displayed...'


--- 
Copy & Paste ready text:

At the time of registration, and for providing communications between the user account and their contacts, the user has to provide a display name (pen name), a username (nickname) and a working email address. The names will be accessible on the profile page of the account by any visitor of the page, even if other profile details are not displayed. The email address will only be used to send the user notifications about interactions, but wont be visibly displayed. The listing of an account in the node's user directory or the global user directory is optional and can be controlled in the user settings, it is not necessary for communication.

**Comment by tobiasd** [comment]
Thanks @AndyHee

File: mod/tos.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * The content of the /tos page is generated from two parts.
+ * (1) a free form part the admin of the node can set in the admin panel
+ * (2) an optional privacy statement that gives some transparency about
+ *     what information are needed by the software to provide the service.
+ *     This privacy statement has fixed text, so it can be translated easily.
+ *
+ * @param App $a
+ * @return string
+ */
+function tos_content(App $a) {
```
Comment (MrPetovan) [inline mod/tos.php]: Can you please move this new module to the `src/Module/` folder by taking example on the new classes already there? The `mod/` folder is destined to disappear altogether as the `include/` folder.

Comment (tobiasd) [reply mod/tos.php]: No. I studied the files and I tried to put the module into src/Module but it does not work. Is there any properly documented example for dummies?

Comment (MrPetovan) [reply mod/tos.php]: The basics: for `/tos` query string, you want a `src/Module/Tos.php` file with a `class Tos extends Module` inside.

Then copy the content of `tos_content()` in `Tos::content()`, remember to add a `$a = self::getApp()` if you are using `$a`.

In the same fashion, copy the content of `tos_init()` in `Tos::init()` with the same caveat.

Comment (tobiasd) [reply mod/tos.php]: Yes, that was the obvious part. The naming was the hidden secret. I added a hint about the naming to the BaseModule for future reference.

Comment (MrPetovan) [reply mod/tos.php]: Thanks!

**Pull request merged**